### PR TITLE
media-gfx/pngnq: Fix passing incompatible pointer type

### PR DESCRIPTION
**Pull request by listout**
Closes: https://bugs.gentoo.org/919210

### CODE REVIEW

File: media-gfx/pngnq/files/pngnq-1.1-gcc14-build-fix.patch [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Bug: https://bugs.gentoo.org/919210
```
Comment (orbea) [inline media-gfx/pngnq/files/pngnq-1.1-gcc14-build-fix.patch:1]: Have you tried to upstream this patch?

File: media-gfx/pngnq/pngnq-1.1-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools
+
+DESCRIPTION="Pngnq is a tool for quantizing PNG images in RGBA format"
+HOMEPAGE="http://pngnq.sourceforge.net/"
```
Comment (mm1ke) [inline media-gfx/pngnq/pngnq-1.1-r2.ebuild:9]:
```suggestion
HOMEPAGE="https://pngnq.sourceforge.net/"
```

File: media-gfx/pngnq/files/pngnq-1.1-gcc14-build-fix.patch [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+--- a/src/pngcomp.c
++++ b/src/pngcomp.c
+@@ -503,7 +503,7 @@ void printstats(struct statistics* stats, struct blockstats* bstats){
+   printf("Mean pixel color error: %f \n",stats->mean_error);
+   printf("Maximum pixel color error: %f \n",stats->max_error);
+   printf("Standard Deviation of error: %f\n",stats->stddev_error);
+-  printf("Image Dimensions %ld x %ld \n",image1_info.width,image1_info.height);
++  printf("Image Dimensions %u x %u \n",image1_info.width,image1_info.height);
```
Comment (N-R-K) [inline media-gfx/pngnq/files/pngnq-1.1-gcc14-build-fix.patch:9]: The correct type for `uint32_t` is PRIu32 macro found in [inttypes.h](https://en.cppreference.com/w/c/types/integer#Format_constants_for_the_fprintf_family_of_functions) header file.

```c
printf("Image Dimensions %"PRIu32" x %"PRIu32" \n",image1_info.width,image1_info.height);
```

Although in this case `png_uint_32` seems to *not* be a typedef to the `stdint.h` types. Just casting it to a type that's known to be big enough should be a portable way to handle cases like this:

```c
printf("Image Dimensions %lld x %lld \n", (long long int)image1_info.width, (long long int)image1_info.height);
```

**Review by N-R-K — CHANGES_REQUESTED** [review]

File: media-gfx/pngnq/files/pngnq-1.1-gcc14-build-fix.patch (hunk 1 above)
Comment (listout) [reply media-gfx/pngnq/files/pngnq-1.1-gcc14-build-fix.patch:1]: Yep, I've also updated the patch with the link.

https://sourceforge.net/p/pngnq/patches/7/

**Review by juippis — APPROVED** [review]
Heh, these ebuilds where 50 % of text are patches to make it work.

**Pull request closed without merging**